inc/saibuffer.h

```diff
@@ -546,6 +546,29 @@ typedef enum _sai_buffer_profile_threshold_mode_t
 
 } sai_buffer_profile_threshold_mode_t;
 
+/**
+ * @brief Enum defining queue actions in case the packet fails to pass the admission control.
+ */
+typedef enum _sai_buffer_profile_packet_admission_fail_action_t
+{
+    /**
+     * @brief Drop the packet.
+     *
+     * Default action. Packet has nowhere to go
+     * and will be dropped.
+     */
+    SAI_BUFFER_PROFILE_PACKET_ADMISSION_FAIL_ACTION_DROP = 0x00000000,
+
+    /**
+     * @brief Trim the packet.
+     *
+     * Try sending a shortened packet over a different
+     * queue. SAI_QUEUE_STAT_DROPPED_PACKETS as well as SAI_QUEUE_STAT_DROPPED_BYTES
+     * will count the original discarded frames even if they will be trimmed afterwards.
+     */
+    SAI_BUFFER_PROFILE_PACKET_ADMISSION_FAIL_ACTION_DROP_AND_TRIM = 0x00000001,
+} sai_buffer_profile_packet_admission_fail_action_t;
+
```

Comment:
Original packet will be dropped and shortened copy of the packet will be send. This need to be clarified in the comments.
interface stats MUST show dropped packets
interface stats MAY show trimmed packets send

Reply:
Added to the description